panel/components/wired/buttons.py

```diff
@@ -0,0 +1,48 @@
+"""Implementation of the wired WebComponent"""
+import param
+
+from panel.pane import WebComponent
+
+
+class RadioButton(WebComponent):
+    """A Wired RadioButton"""
+    html = param.String('<wired-radio checked id="1" checked>Radio Two</wired-radio>')
+
+class CheckBox(WebComponent):
+    html = param.String('<wired-checkbox id="strokeShape" checked="" class="wired-rendered">Stroke shape</wired-checkbox>')
+
+class Slider(WebComponent):
+    html = param.String('<wired-slider id="slider" value="33" knobradius="15" class="wired-rendered" style="margin: 0px"></wired-slider>')
+    def __init__(self, min_height=40, **params):
+        super().__init__(min_height=min_height, **params)
+
+class ComboBox(WebComponent):
+    html = param.String("""<wired-combo id="colorCombo" selected="red" role="combobox" aria-haspopup="listbox" tabindex="0" class="wired-rendered" aria-expanded="false"><wired-item value="red" aria-selected="true" role="option" class="wired-rendered">Red</wired-item><wired-item value="green" role="option" class="wired-rendered">Green</wired-item><wired-item value="blue" role="option" class="wired-rendered">Blue</wired-item></wired-combo>""")
+    def __init__(self, min_height=40, **params):
+        super().__init__(min_height=min_height, **params)
+
+class Video(WebComponent):
+    html = param.String("""<wired-video autoplay="" playsinline="" muted="" loop="" style="height: 100%; margin: 0px" src="https://file-examples.com/wp-content/uploads/2017/04/file_example_MP4_480_1_5MG.mp4" class="wired-rendered"></wired-video>""")
+
+if __name__.startswith("bk"):
+    import panel as pn
+    pn.config.sizing_mode="stretch_width"
+
+    js = """

```

Comment:
Would be nicer to append these urls to `pn.config.js_files`.

Reply:
Yes. Eventually they should.

If you take a look the value of `js` you can see only one of the two imports can be done via `pn.config.js_files`. In one of the imports I need to specify the `type`.

```python
js = """
<script src="https://unpkg.com/@webcomponents/webcomponentsjs@2.0.0/webcomponents-loader.js"></script>
<script type="module" src="https://unpkg.com/wired-elements@0.6.4/dist/wired-elements.bundled.js"></script>
"""
```

In general the import of `js` web component libraries does not always work for me. It needs some experimentation before it works. Sometimes I need to 

- add `type="module"`
- add ``?module` to the url.
- add additional javascript libraries than what is described on npm or in the documentation.

(I believe)

So I guess there will be a request to improve the `pn.config.js_files` at some stage. At a minimum be able to specify the `type`.

Reply:
I was wondering about that. Will have to do some reading before making any concrete suggestions.